lib/services/api/href_parser.rb

```diff
@@ -0,0 +1,72 @@
+module Api
+  class HrefParser
+    def self.parse(href)
```

Comment:
Mainly because I wanted a way to express a very narrow public API for this service, which is a bit more gracefully expressed as an instance. I can also anticipate some further work being done here where instead of an `HrefParser` it might simply embody an `Href` (or `MiqHref`) so we could ask it some questions that are currently expressed private methods. We duplicate some of this knowledge in various places (see, for example, https://github.com/ManageIQ/manageiq-api/blob/master/app/controllers/api/base_controller/parser/request_adapter.rb).

Reply:
Cool, makes sense.  Works for me, and a single object allocation on each request probably isn't going to cause problems here.

My comment was more of a "is there a reason for this that I am missing", and less of a "you should change this" one (though that probably wasn't clear).

Reply:
@NickLaMuro FWIW this is a really common pattern that I preach. ([example](https://github.com/ManageIQ/manageiq/pull/16023#pullrequestreview-64897892)) Looks great to me.